backend/plugin/parser/base/transaction_mode.go

```diff
@@ -1,33 +1,94 @@
 package base
 
 import (
+	"database/sql"
 	"regexp"
 	"strings"
 
 	"github.com/bytebase/bytebase/backend/common"
 )
 
-// txnModeDirectiveRegex matches the transaction mode directive at the beginning of a SQL script.
-// Format: -- txn-mode = on|off
-var txnModeDirectiveRegex = regexp.MustCompile(`(?i)^\s*--\s*txn-mode\s*=\s*(on|off)\s*$`)
+// Directive regex patterns
+var (
+	// txnModeDirectiveRegex matches the transaction mode directive.
+	// Format: -- txn-mode = on|off
+	txnModeDirectiveRegex = regexp.MustCompile(`(?i)^\s*--\s*txn-mode\s*=\s*(on|off)\s*$`)
+
+	// txnIsolationDirectiveRegex matches the isolation level directive.
+	// Format: -- txn-isolation = READ UNCOMMITTED|READ COMMITTED|REPEATABLE READ|SERIALIZABLE
+	txnIsolationDirectiveRegex = regexp.MustCompile(`(?i)^\s*--\s*txn-isolation\s*=\s*(READ\s+UNCOMMITTED|READ\s+COMMITTED|REPEATABLE\s+READ|SERIALIZABLE)\s*$`)
+)
 
 // ParseTransactionMode extracts the transaction mode directive from the SQL script.
 // It checks the first line of the script for the -- txn-mode = on|off directive.
 // Returns the transaction mode and the SQL script without the directive.
+// Deprecated: Use ParseTransactionConfig instead for full transaction configuration support.
 func ParseTransactionMode(script string) (common.TransactionMode, string) {
+	config, cleanScript := ParseTransactionConfig(script)
+	return config.Mode, cleanScript
+}
+
+// ParseTransactionConfig extracts both transaction mode and isolation level directives from the SQL script.
+// It checks the first two lines for directives:
+// - Line 1: -- txn-mode = on|off
```

Comment:
Let's be less restrictive on the line / parameter mapping. Users should be able to write txn-mode/txn-isolation on any line on the top of the file at least.